example/spec/parallel/discoverGranules/DiscoverGranulesDuplicateHandlingSkipSpec.js

```diff
@@ -0,0 +1,206 @@
+      provider = await createProvider(prefix, { host: sourceBucket });
+
+      // Stage the existing granule file to S3
+      existingGranuleId = randomId('existing-granule-');
+      existingGranuleKey = `${sourcePath}/${existingGranuleId}.txt`;
+      await s3PutObject({
+        Bucket: sourceBucket,
+        Key: existingGranuleKey,
+        Body: 'asdf'
+      });
+
+      // Ingest the existing granule
+      ingestGranuleRule = await createOneTimeRule(
+        prefix,
+        {
+          workflow: 'IngestGranule',
+          collection: pick(collection, ['name', 'version']),
+          provider: provider.id,
+          payload: {
+            testExecutionId: randomId('test-execution-'),
+            granules: [
+              {
+                granuleId: existingGranuleId,
+                dataType: collection.name,
+                version: collection.version,
+                files: [
+                  {
+                    name: `${existingGranuleId}.txt`,
+                    path: sourcePath
+                  }
+                ]
+              }
+            ]
+          }
+        }
+      );
+
+      // Find the "IngestGranule" execution ARN
+      const ingestGranuleExecutionArn = await findExecutionArn(
+        prefix,
+        (execution) =>
+          get(execution, 'originalPayload.testExecutionId') === ingestGranuleRule.payload.testExecutionId,
+        { timeout: 15 }
+      );
+
+      // Wait for the "IngestGranule" execution to be completed
+      await getCompletedExecution({ prefix, arn: ingestGranuleExecutionArn });
+
+      // Wait for the existing granule to be fully ingested
+      await getCompletedGranule({ prefix, granuleId: existingGranuleId });
+
+      // Stage the new granule file to S3
+      newGranuleId = randomId('new-granule-');
+      newGranuleKey = `${sourcePath}/${newGranuleId}.txt`;
+      await s3PutObject({
+        Bucket: sourceBucket,
+        Key: newGranuleKey,
+        Body: 'asdf'
+      });
+
+      // Run DiscoverGranules
+      discoverGranulesRule = await createOneTimeRule(
+        prefix,
+        {
+          workflow: 'DiscoverGranules',
+          collection: {
+            name: collection.name,
+            version: collection.version
+          },
+          provider: provider.id,
+          payload: {
+            testExecutionId: randomId('test-execution-')
+          }
+        }
+      );
+
+      // Find the "DiscoverGranules" execution ARN
+      const discoverGranulesExecutionArn = await findExecutionArn(
+        prefix,
+        (execution) =>
+          get(execution, 'originalPayload.testExecutionId') === discoverGranulesRule.payload.testExecutionId,
+        { timeout: 15 }
+      );
+
+      // Get the completed "DiscoverGranules" execution
+      finishedDiscoverGranulesExecution = await getCompletedExecution({
+        prefix,
+        arn: discoverGranulesExecutionArn
+      });
+    } catch (err) {
+      beforeAllFailed = true;
```

Comment:
It would probably be preferable to set `beforeAllFailed = err` and then call `fail(beforeAllFailed)` so that we know why the test failed from the beforeAll